### ISSUE TITLE
Is multithreading supported?

### ISSUE THREAD

**Issue by FreeVB** [fadymaher24/AsyncTcpServer#1]
Is multithreading supported?

**Comment by fadymaher24** [comment]
Yes, the code is multithreading supported. It uses asynchronous programming with the async/await pattern, allowing it to efficiently handle multiple client connections concurrently without blocking threads. Each client connection is processed in its own asynchronous task, making it suitable for a multithreaded environment where multiple clients can be served simultaneously.

**Comment by FreeVB** [comment]
Can you write a multithreaded demo? Thank you.
